beacon_node/network/src/sync/block_lookups/common.rs

```diff
@@ -380,18 +380,24 @@ impl<L: Lookup, T: BeaconChainTypes> RequestState<L, T> for BlobRequestState<L,
                 let received_id = blob.id();
                 if !self.requested_ids.contains(&received_id) {
                     self.state.register_failure_downloading();
-                    Err(LookupVerifyError::UnrequestedBlobId)
-                } else {
-                    // State should remain downloading until we receive the stream terminator.
-                    self.requested_ids.remove(&received_id);
-                    let blob_index = blob.index;
-
-                    if blob_index >= T::EthSpec::max_blobs_per_block() as u64 {
-                        return Err(LookupVerifyError::InvalidIndex(blob.index));
-                    }
-                    *self.blob_download_queue.index_mut(blob_index as usize) = Some(blob);
-                    Ok(None)
+                    return Err(LookupVerifyError::UnrequestedBlobId);
+                }
+
+                blob.verify_blob_sidecar_inclusion_proof()
+                    .map_err(|_| LookupVerifyError::InvalidInclusionProof)?;
+                if blob.block_root() != expected_block_root {
+                    return Err(LookupVerifyError::UnrequestedHeader);
+                }
+
+                // State should remain downloading until we receive the stream terminator.
+                self.requested_ids.remove(&received_id);
+                let blob_index = blob.index;
+
+                if blob_index >= T::EthSpec::max_blobs_per_block() as u64 {
+                    return Err(LookupVerifyError::InvalidIndex(blob.index));
                 }
```

Comment:
Suggestion on ordering:
- I think we'd want to update the `self.requested_ids` last, only if we are not returning any errors.
- Perhaps validate the `blob.index` and `blob.block_root` before verifying inclusion proof since it's probably faster to verify them?

```suggestion
                if blob.block_root() != expected_block_root {
                    return Err(LookupVerifyError::UnrequestedHeader);
                }
                
                let blob_index = blob.index;
                if blob_index >= T::EthSpec::max_blobs_per_block() as u64 {
                    return Err(LookupVerifyError::InvalidIndex(blob.index));
                }

                blob.verify_blob_sidecar_inclusion_proof()
                    .map_err(|_| LookupVerifyError::InvalidInclusionProof)?;
                    
                // State should remain downloading until we receive the stream terminator.
                self.requested_ids.remove(&received_id);
```

Reply:
we could also just drop the blob index check since it'd be covered by the inclusion proof verification